test/integration/good/code-gen/cpp.expected

```diff
@@ -20110,6 +20110,573 @@ stan::model::model_base& new_model(
+(void) function__;  // suppress unused var warning
+
+    stan::io::reader<local_scalar_t__> in__(params_r__, params_i__);
+    local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+    (void) DUMMY_VAR__;  // suppress unused var warning
+
+    
+    try {
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>> L_Omega;
+      L_Omega = std::vector<Eigen::Matrix<local_scalar_t__, -1, -1>>(nt, Eigen::Matrix<local_scalar_t__, -1, -1>(2, 2));
+      stan::math::fill(L_Omega, DUMMY_VAR__);
+      
+      std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>> L_Omega_in__;
+      L_Omega_in__ = std::vector<Eigen::Matrix<local_scalar_t__, -1, 1>>(nt, Eigen::Matrix<local_scalar_t__, -1, 1>(
+        ((2 * (2 - 1)) / 2)));
+      stan::math::fill(L_Omega_in__, DUMMY_VAR__);
+      
+      current_statement__ = 1;
+      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
+        current_statement__ = 1;
+        assign(L_Omega_in__, cons_list(index_uni(sym1__), nil_index_list()),
+          in__.vector(((2 * (2 - 1)) / 2)), "assigning variable L_Omega_in__");
+      }
+      current_statement__ = 1;
+      for (int sym1__ = 1; sym1__ <= nt; ++sym1__) {
+        current_statement__ = 1;
+        if (jacobian__) {
+          current_statement__ = 1;
+          assign(L_Omega, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)],
+              2, lp__), "assigning variable L_Omega");
+        } else {
+          current_statement__ = 1;
+          assign(L_Omega, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::cholesky_corr_constrain(L_Omega_in__[(sym1__ - 1)], 2),
+            "assigning variable L_Omega");
+        }}
+      Eigen::Matrix<local_scalar_t__, -1, 1> z1;
+      z1 = Eigen::Matrix<local_scalar_t__, -1, 1>(NS);
+      stan::math::fill(z1, DUMMY_VAR__);
+      
+      current_statement__ = 2;
+      z1 = in__.vector(NS);
+      current_statement__ = 2;
+      for (int sym1__ = 1; sym1__ <= NS; ++sym1__) {
+        current_statement__ = 2;
+        if (jacobian__) {
+          current_statement__ = 2;
+          assign(z1, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::lb_constrain(z1[(sym1__ - 1)],
+              rvalue(L_Omega,
+                cons_list(index_uni(1),
+                  cons_list(index_uni(1),
+                    cons_list(index_uni(2), nil_index_list()))), "L_Omega"),
+              lp__), "assigning variable z1");
+        } else {
+          current_statement__ = 2;
+          assign(z1, cons_list(index_uni(sym1__), nil_index_list()),
+            stan::math::lb_constrain(z1[(sym1__ - 1)],
+              rvalue(L_Omega,
+                cons_list(index_uni(1),
+                  cons_list(index_uni(1),
+                    cons_list(index_uni(2), nil_index_list()))), "L_Omega")),
+            "assigning variable z1");
+        }}
```

Comment:
Is there a reason a lot of the vectorized versions of this code is ignored? Like here again reader has [`vector_lb_constrain()`](https://github.com/stan-dev/stan/blob/develop/src/stan/io/reader.hpp#L1304)

I didn't know this is what actually happened in constrain etc functions within the compiler. For the new matrix type we really need to use those vectorized specializations or this will be a performance killer